UnitTests/Onboarding/ContextualOnboarding/BrowserTabViewControllerOnboardingTests.swift

```diff
@@ -56,9 +58,20 @@ final class BrowserTabViewControllerOnboardingTests: XCTestCase {
         viewController = nil
         cancellables = []
         expectation = nil
+        featureFlagger = nil
         try super.tearDownWithError()
     }
 
+    func testWhenNavigationCompletedAndFeatureIsOffThenTurnOffFeature() throws {
+        featureFlagger.isFeatureOn = false
+        let expectation = self.expectation(description: "Wait for turnOffFeatureCalled to be called")
+        dialogProvider.dialogTypeForTabExpectation = expectation
```

Comment:
you meant to use `turnOffFeatureCalledExpectation` for dialogProvider?

Reply:
yes cause here conforms to both the dialogProvider and stateUpdater protocols

Reply:
sorry I misread yes you are right…